sdks/python/apache_beam/io/filesystem_test.py

```diff
@@ -514,25 +514,25 @@ def create_test_file(compression_type, lines):
     # different error, in the Beam 2.13 and earlier implementation.
     # So it's not strictly necessary to have this handler in this unit test.
 
-    def alarm_handler(signum, frame):
-      self.fail('Timed out reading compressed file.')
+    def timeout_handler():
       raise IOError('Exiting due to likley infinite loop logic in code.')
 
-    old_handler = signal.signal(signal.SIGALRM, alarm_handler)
+    timer = threading.Timer(timeout, timeout_handler)
     try:
       test_lines = tuple(generate_random_line() for i in range(num_test_lines))
       for compression_type in [CompressionTypes.BZIP2, CompressionTypes.GZIP]:
         file_name = create_test_file(compression_type, test_lines)
-        signal.alarm(timeout)
+        timer.start()
         with open(file_name, 'rb') as f:
           data = CompressedFile(f, compression_type, read_size=read_size)
           for written_line in test_lines:
             read_line = data.readline()
             self.assertEqual(written_line, read_line)
-        signal.alarm(0)
+        timer.cancel()
+        # Starting a new timer for the next iteration/test.
```

Comment:
nit: can't we create the timer at the beginning of the for loop or do we want to include the test line generation to be part of it.

New flow would be:
```
generate test lines
for ...:
  timer = ...
  file_name = ...
  ...
  timer.cancel()
```

This would allow us to remove the finally block.

Reply:
I think it's good to keep timer in a finally block to make sure it shuts down in all cases.

Reply:
makes sense.